src/main/resources/assets/admin/common/js/app/browse/BrowsePanel.ts

```diff
@@ -104,26 +105,65 @@ export class BrowsePanel
     private initTreeGridListeners() {
         this.selectableListBoxPanel?.onDataChanged(this.handleDataChanged.bind(this));
 
-        const selectionChangedListener = AppHelper.debounce(this.handleTreeListSelectionChanged.bind(this), 250);
+        let recentlySelected = false;
+        let recentlySelectedTimeout = 0;
+        // Every selection change fires deselect and select events
+        // Use 50ms debounce to prevent handling deselect event
+        // Keep debounce time minimal as it affects user perception after selecting single content
+        const selectionChangedListener = AppHelper.debounce((selection) => {
+            let delayedHandler = true;
+            if (!recentlySelected) {
+                this.handleTreeListSelectionChanged(selection);
+                delayedHandler = false;
+            }
+
+            recentlySelected = true;
+
+            if (recentlySelectedTimeout) {
+                clearTimeout(recentlySelectedTimeout);
+            }
+            recentlySelectedTimeout = window.setTimeout(() => {
+                recentlySelected = false;
+                if (delayedHandler) {
+                    this.handleTreeListSelectionChanged(selection);
+                }
+            }, 250);
+
+        }, 50);
+
         this.selectableListBoxPanel?.onSelectionChanged(selectionChangedListener);
 
+        if (this.keyNavigator) {
+            const upDownHandler = function () {
+                // mark recently selected content before selectionChangedListener reads it
+                // to prevent loading content when scrolling with keyboard
+                if (!recentlySelected) {
+                    recentlySelected = true;
+                }
+                return false;
+            }
+            this.keyNavigator.onKeyDown(upDownHandler);
+            this.keyNavigator.onKeyUp(upDownHandler);
+        }
```

Comment:
Instead of using `keyNavigator` in the base `BrowsePanel` class we should change `initTreeGridListeners` to be protected and add the additional logic (where `keyNavigator` is used) in Content Studio's `ContentBrowsePanel`. `recentlySelected` will then have to be made a protected class property as well.

Reply:
Actually, `recentlySelected` should be private and there should be protected methods called `setRecentlySelected` and `resetRecentlySelected` in BrowsePanel

Reply:
Well, having it in BrowsePanel expands that functionality to all descendants.

If you on the other hand make `initTreeGridListeners` protected and put the logic to the ContentBrowsePanel, we're going to only have it ContentBrowsePanel.

Since keyNavigator is of type `SelectableTreeListBoxKeyNavigator` that comes from lib-admin-ui and is not exclusive to content studio, it makes perfect sense to have it in the base class.

And finally, pattern with using protected methods implemented by descendants is widely used in those classes so I just don't see a reason we can't use it here ?

Reply:
```    
    protected updateFilterPanelOnSelectionChange() {
        // TO BE IMPLEMENTED BY INHERITORS
    }

    protected enableSelectionMode() {
        // implement in inheritors
    }

    protected disableSelectionMode() {
        // implement in inheritors
    }

    protected createToolbar(): Toolbar<ToolbarConfig> {
        throw Error('Must be implemented by inheritors');
    }

    protected createListBoxPanel(): SelectableListBoxPanel<ViewItem> {
        throw Error('Must be implemented by inheritors');
    }

    protected createBrowseItemPanel(): BrowseItemPanel {
        throw Error('Must be implemented by inheritors');
    }
```